astroquery/heasarc/core.py

```diff
@@ -15,31 +17,158 @@
 class HeasarcClass(BaseQuery):
 
     """HEASARC query class.
+
+    For a list of available HEASARC mission tables, visit:
+        https://heasarc.gsfc.nasa.gov/cgi-bin/W3Browse/w3catindex.pl
     """
 
     URL = conf.server
     TIMEOUT = conf.timeout
 
-    def query_object_async(self, object_name, mission, cache=True,
-                           get_query_payload=False,
-                           display_mode='FitsDisplay'):
-        """TODO: document this!
+    def query_async(self, request_payload, cache=True, url=conf.server):
+        """
+        Submit a query based on a given request_payload. This allows detailed
+        control of the query to be submitted.
+        """
+        response = self._request('GET', url, params=request_payload,
+                                 timeout=self.TIMEOUT, cache=cache)
+        return response
 
-        (maybe start by copying over from some other service.)
+    def query_mission_list(self, cache=True, get_query_payload=False):
+        """
+        Returns a list of all available mission tables with descriptions
         """
-        request_payload = dict()
-        request_payload['Entry'] = object_name
-        request_payload['tablehead'] = ('BATCHRETRIEVALCATALOG_2.0 {}'
-                                        .format(mission))
-        request_payload['Action'] = 'Query'
-        request_payload['displaymode'] = display_mode
+        request_payload = self._args_to_payload(
+            Entry='none',
+            mission='xxx',
+            displaymode='BatchDisplay'
+        )
 
         if get_query_payload:
             return request_payload
 
-        response = self._request('GET', self.URL, params=request_payload,
-                                 timeout=self.TIMEOUT, cache=cache)
-        return response
+        # Parse the results specially (it's ascii format, not fits)
+        response = self.query_async(
+            request_payload,
+            url='https://heasarc.gsfc.nasa.gov/db-perl/W3Browse/w3query.pl',
```

Comment:
Should this be a configuration item?

Reply:
I've moved this into 'heasarc/__init__.py' as a configuration parameter.